src/sentry/static/sentry/app/views/settings/incidentRules/triggers/actionsPanel/index.tsx

```diff
@@ -63,37 +65,65 @@ const getPlaceholderForType = (type: ActionType) => {
       return 'username or channel';
     case ActionType.PAGERDUTY:
       return 'service';
+    case ActionType.INTEGRATION:
+      return 'integration';
     default:
       throw Error('Not implemented');
   }
 };
+
 /**
- * Lists saved actions as well as control to add a new action
+ * When a new action is added, all of it's settings should be set to their default values.
+ * @param actionConfig
  */
-class ActionsPanel extends React.PureComponent<Props> {
-  /**
-   * Actions have a type (e.g. email, slack, etc), but only some have
-   * an integrationId (e.g. email is null). This helper creates a unique
-   * id based on the type and integrationId so that we know what action
-   * a user's saved action corresponds to.
-   */
-  getActionUniqueKey({type, integrationId}: Pick<Action, 'type' | 'integrationId'>) {
-    return `${type}-${integrationId}`;
-  }
+const getCleanAction = (actionConfig): Action => {
```

Comment:
So I ran into a couple of problems here when testing:
* `targetType` is null - because we don't actually have any `allowedTargetTypes` for sentry apps, because we don't need them, this returns null and it's required.
* `targetIdentifier` not specified - this is also required, and I think we settled on this being the `sentry_app_id` which in this case would be the `actionConfig.integrationId`

Normally the `targetType` and the `targetIdentifier` would get set when you change the target. Since we don't have that, we are going to need to set those fields when the action type itself changes. Pulling it out into this function is great cause we can just do that logic in here